### PR TITLE
Fire 'select' event in SetSelection{Start,End}

### PR DESCRIPTION
**Pull request by jonleighton**
See #19171.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19202)

### CODE REVIEW

File: tests/wpt/metadata/html/semantics/forms/textfieldselection/selection-start-end.html.ini [hunk 1]
```diff
@@ -17,26 +17,26 @@
...
-    expected: NOTRUN
+    expected: PASS
 
   [onselect should fire when selectionStart is changed on input-not-appended]
-    expected: NOTRUN
+    expected: PASS
 
   [onselect should fire when selectionStart is changed on input-appended-prefocused]
-    expected: NOTRUN
+    expected: PASS
```
Comment (tigercosmos) [inline tests/wpt/metadata/html/semantics/forms/textfieldselection/selection-start-end.html.ini]: If Pass, You can delete it directly

**Review by tigercosmos — CHANGES_REQUESTED** [review]
You can delete the pass test cases in  `tests/wpt/metadata/html/semantics/forms/textfieldselection/selection-start-end.html.ini` 
We only need to remain the unexpected ones

File: components/script/dom/htmlinputelement.rs [hunk 2]
```diff
@@ -886,6 +873,23 @@ impl HTMLInputElement {
             _ => ()
         }
     }
+
+    fn selection_direction(&self) -> SelectionDirection {
+        self.textinput.borrow().selection_direction
+    }
+
+    fn set_selection_range(&self, start: u32, end: u32, direction: SelectionDirection) {
```
Comment (KiChjang) [inline components/script/dom/htmlinputelement.rs]: Let's add a spec link comment above this function (https://html.spec.whatwg.org/multipage/#set-the-selection-range).

File: components/script/dom/htmlinputelement.rs [hunk 3]
```diff
@@ -886,6 +873,23 @@ impl HTMLInputElement {
...
+        self.textinput.borrow_mut().selection_direction = direction;
+        self.textinput.borrow_mut().set_selection_range(start, end);
+        let window = window_from_node(self);
+        let _ = window.user_interaction_task_source().queue_event(
```
Comment (KiChjang) [inline components/script/dom/htmlinputelement.rs]: Let's also add step annotations for this function. For example, this particular line here follows step 6 of the linked algorithm (albeit incorrectly).

**Review by KiChjang — CHANGES_REQUESTED** [review]
👍 Great start! I only have minor comments about adding comments ;). Once you've addressed them, you can r=me.

**Comment by jonleighton** [comment]
Thanks, requested changes implemented

File: tests/wpt/metadata/html/semantics/forms/textfieldselection/selection-start-end.html.ini [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [selection-start-end.html]
   type: testharness
-  expected: TIMEOUT
+  expected: OK
```
Comment (KiChjang) [inline tests/wpt/metadata/html/semantics/forms/textfieldselection/selection-start-end.html.ini]: You can remove this line now.

**Review by KiChjang — CHANGES_REQUESTED** [review]
One final nit!

**Comment by KiChjang** [comment]
Also, after you've made the changes, can you squash your commits into one, please?

**Comment by jonleighton** [comment]
Sure, done

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Pull request merged**